app/services/proforma_service/convert_task_to_exercise.rb

```diff
@@ -21,17 +21,24 @@ def import_exercise
         user: @user,
         title: @task.title,
         description: @task.description,
-        public: @task.meta_data[:CodeOcean]&.dig(:public) == 'true',
-        hide_file_tree: @task.meta_data[:CodeOcean]&.dig(:hide_file_tree) == 'true',
-        allow_file_creation: @task.meta_data[:CodeOcean]&.dig(:allow_file_creation) == 'true',
-        allow_auto_completion: @task.meta_data[:CodeOcean]&.dig(:allow_auto_completion) == 'true',
+        public: string_to_bool(@task.meta_data[:CodeOcean]&.dig(:public)),
+        hide_file_tree: string_to_bool(@task.meta_data[:CodeOcean]&.dig(:hide_file_tree)),
+        allow_file_creation: string_to_bool(@task.meta_data[:CodeOcean]&.dig(:allow_file_creation)),
+        allow_auto_completion: string_to_bool(@task.meta_data[:CodeOcean]&.dig(:allow_auto_completion)),
         expected_difficulty: @task.meta_data[:CodeOcean]&.dig(:expected_difficulty),
         execution_environment_id: @task.meta_data[:CodeOcean]&.dig(:execution_environment_id),
 
         files: files
       )
     end
 
+    def string_to_bool(str)
+      return true if str == 'true'
+      return false if str == 'false'
+
+      nil
+    end
```

Comment:
Alternatively, we could use `ActiveModel::Type::Boolean.new.cast(value)`

Reply:
We want to return `nil`, which is not available by the aforementioned `cast` method. We keep it.